jenkinsfiles/Jenkinsfile.grobid

```diff
@@ -17,8 +17,16 @@ elifePipeline {
 
         stage "Build Image", {
             dir('grobid') {
+                def dockerfile = "Dockerfile"
+                if (new File("Dockerfile.crf")) {
```

Comment:
I'm not sure what is going on with Groovy and Java's `java.io.File` object here, there seem to be two sets of documentation ([Java](https://docs.oracle.com/javase/8/docs/api/java/io/File.html) and [Groovy](http://docs.groovy-lang.org/latest/html/groovy-jdk/java/io/File.html)). I *think* the Groovy documentation is for the convenience methods it's adding on top of what is in `File` rather than providing a completely new `File` object. The 'overview' page [hints at this](http://docs.groovy-lang.org/latest/html/groovy-jdk/overview-summary.html).

If the above is correct then that `new File("Dockerfile.crf")` will create a `File` object regardless of whether `"Dockerfile.crf"` exists or not and `if (new File("Dockerfile.crf"))` will **always** evaluate to true.

You probably want to use `.exists` or `.isFile`, so something like `if (new File("Dockerfile.crf").isFile()) { ... }` (you may be able to drop that empty ellipsis in Groovy, not sure).

I could be completely wrong though, Groovy introduces so much magic.

Reply:
You are right, I was meant to add `.exists()` but then forgot. Could that you spotted that, should have spotted it myself.